File: benchmark/mandelbrot/mandelbrot.cs
using System;

static class Bench
{
    static void Main()
    {
        const int width = 256;
        const int maxIter = 64;
        const double step = 2.0 / 256.0;

        int inside = 0;
        for (int py = 0; py < width; py++)
        {
            double cy = -1.0 + py * step;
            for (int px = 0; px < width; px++)
            {
                double cx = -1.5 + px * step;
                double zr = 0.0;
                double zi = 0.0;
                int iter = 0;
                while (iter < maxIter && zr * zr + zi * zi <= 4.0)
                {
                    double t = zr * zr - zi * zi + cx;
                    zi = 2.0 * zr * zi + cy;
                    zr = t;
                    iter += 1;
                }
                if (iter == maxIter)
                {
                    inside += 1;
                }
            }
        }
        Console.WriteLine(inside);
    }
}

File: benchmark/mandelbrot/mandelbrot.py
WIDTH = 256
MAX_ITER = 64
STEP = 2.0 / 256.0

inside = 0
for py in range(WIDTH):
    cy = -1.0 + py * STEP
    for px in range(WIDTH):
        cx = -1.5 + px * STEP
        zr = 0.0
        zi = 0.0
        it = 0
        while it < MAX_ITER and zr * zr + zi * zi <= 4.0:
            zr, zi = zr * zr - zi * zi + cx, 2.0 * zr * zi + cy
            it += 1
        if it == MAX_ITER:
            inside += 1
print(inside)
